lib/ws-kernel.js

```diff
@@ -53,66 +54,35 @@ export default class WSKernel extends Kernel {
         this._callWatchCallbacks();
       }
 
-      if (onResults) {
-        log("WSKernel: _execute:", message);
-        const result = this._parseIOMessage(message);
-        if (result) onResults(result);
-      }
-    };
-
-    future.onReply = (message: Message) => {
-      const result = {
-        data: message.content.status,
-        stream: "status"
-      };
-      if (onResults) onResults(result);
-    };
-
-    future.onStdin = (message: Message) => {
-      if (message.header.msg_type !== "input_request") {
-        return;
-      }
-
-      const { prompt } = message.content;
-
-      const inputView = new InputView({ prompt }, (input: string) =>
-        this.session.kernel.sendInputReply({ value: input })
-      );
-
-      inputView.attach();
+      log("WSKernel: execute:", message);
+      onResults(message, "iopub");
```

Comment:
`if (onResults) onResults(message, "iopub");`

Reply:
Hydrogen has a setting to run code on startup (we don't Hydrogen to handle the responses from executing the startup code).

Reply:
I handle this by having `_executeStartupCode` specify a dummy results callback `(message, channel) => {}`.

Wouldn't a one-time dummy callback be cleaner than having `if (onResults)` throughout the entire codebase?

Reply:
Wouldn't a one-time `if (onResults)` be cleaner than having if a dummy callback throughout the entire codebase?

---

anyway, not a blocking issue.

Reply:
It's not one-time thing when your next two comments on this PR ask me to change the same thing 😛

Reply:
Just to be clear (because my previous comment was too subtle for its own good): this is a question of a change in the API change vs changes in API-user's code.